tests/src/python/test_qgsvectorlayer.py

```diff
@@ -1619,6 +1619,18 @@ def testUniqueStringsMatching(self):
         self.assertEqual(set(layer.uniqueStringsMatching(0, 'n')),
                          set(['orange', 'BanaNa', 'waterMelon', 'pineapple', 'coconut']))
 
+    def onSubsetStringChanged(self):
+        self.subsetStringChanged = True
+
+    def test_subsetString(self):
+        self.subsetSringChanged = False
+        myPath = os.path.join(unitTestDataPath(), 'lines.shp')
+        layer = QgsVectorLayer(myPath, 'test', 'ogr')
+        layer.subsetStringChanged.connect(self.onSubsetStringChanged)
```

Comment:
Nitpicking here, but given that there's a typo in the variable names here you'll need to revisit this anyway ;)

you could use a local function

```
def test_subsetString(self):
    subset_string_changed = False
    def onSubsetStringChanged():
            subset_string_changed = True
    ...
```